polaris/hub/storage.py

```diff
@@ -125,6 +129,149 @@ def _multipart_upload(self, key: str, value: bytes) -> None:
                 Bucket=self.bucket_name, Key=full_key, UploadId=upload_id, MultipartUpload={"Parts": parts}
             )
 
+    ## Custom methods
+
+    def copy_to_destination(
+        self, destination: Store, if_exists: ZarrConflictResolution = "replace", log: Callable = lambda: None
+    ) -> tuple[int, int, int]:
+        """
+        Copy the content of this store to the destination store.
+
+        We leverage the internal knowledge of this store to make the operation more efficient than `zarr.copy_store`:
+            - Parallel, concurrent `getitems` operations using a thread pool
+
+        Zarr V3 supports partial writes, that would allow us to stream the response back into the store. Unfortunately,
+        it's not supported right now by the library version we use, so we'll have to read the whole response into memory
+        and write it back to the destination.
+
+        Args:
+            destination: destination store
+            if_exists: behavior if the destination key already exists
+            log: optional logging function
+        """
+
+        destination_store_version = getattr(destination, "_store_version", 2)
+        if destination_store_version != self._store_version:
+            raise ValueError("zarr stores must share the same protocol version")
```

Comment:
Maybe we could create a custom exception since it's duplicated.

Reply:
Let's revisit when Zarr V3 is officially out. This code will change then.